pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -232,9 +232,10 @@ func remediate(r *ReconcileMachineHealthCheck, machine *mapiv1.Machine) (reconci
 			durationUnhealthy.String(),
 		)
 
+		unhealthyTimeout := conditionTimeout - durationUnhealthy + time.Second
```

Comment:
Can you more elaborate on why this is the right formula this time?

Reply:
done